library/src/main/java/com/novoda/downloadmanager/DownloadManager.java

```diff
@@ -122,34 +120,22 @@ public void removeDownloadBatchCallback(DownloadBatchCallback downloadBatchCallb
     }
 
     @Override
-    public void getAllDownloadBatchStatuses(AllBatchStatusesCallback callback) {
+    public void getAllDownloadBatchStatuses(final AllBatchStatusesCallback callback) {
         if (downloadService == null) {
-            ensureDownloadServiceExistsAndGetAllDownloadBatchStatuses(callback);
+            executor.submit(
+                    WaitForLockRunnable.waitFor(waitForDownloadService)
+                            .thenPerform(new WaitForLockRunnable.Action() {
+                                @Override
+                                public void performAction() {
+                                    executeGetAllDownloadBatchStatuses(callback);
```

Comment:
shall we execute this back in the main thread?

Reply:
Errrrrr....this wasn't done before? 😂